src/ui/controls/edit.cpp

```diff
@@ -89,7 +89,8 @@ bool IsSep(int character)
 //! Object's constructor.
 CEdit::CEdit()
     : CControl(),
-      m_text( CHARACTERS_MAX + 1, '\0' ),
+      m_maxChar(1000000),
```

Comment:
Now that you fixed the memory allocation, you should be able to use `std::numeric_limits<int>::max()` here